elastic-agent-client.proto

```diff
@@ -9,86 +9,87 @@ package proto;
 option cc_enable_arenas = true;
 option go_package = "pkg/proto;proto";
 
-// Connection information sent to the application on startup so it knows how to connected back to the Elastic Agent.
-//
-// This is normally sent through stdin and should never be sent across a network un-encrypted.
-message ConnInfo {
-    // GRPC connection address.
-    string addr = 1;
-    // Server name to use when connecting over TLS.
-    string server_name = 2;
-    // Token that the application should send as the unique identifier when connecting over the GRPC.
-    string token = 3;
-    // CA certificate.
-    bytes ca_cert = 4;
-    // Peer certificate.
-    bytes peer_cert = 5;
-    // Peer private key.
-    bytes peer_key = 6;
+// Empty response message.
+message EmptyResponse {}
+
+// State codes for the current state.
+enum State {
+    STARTING = 0;
```

Comment:
We will add state diagram to better define that.

Reply:
Would be great to have a diagram of all the "objects" and dependencies. Would make it easier to an overview over the full protocol.

Reply:
It seems to me that this list is the common states of both inputs and outputs. However, I would like to provide a more granular state reporting for inputs. For example, I would like to report if an input is actively publishing events or it is idle. Where should those states go? Are you open to extend this list with unit specific states?

Reply:
@kvch I have a feeling that will result in a lot of flipped states from publish to not published. All of these state changes will be reported to Fleet so it can be exposed and viewable. I would think that providing something that granular and the fact that it would probably flip back and forth a lot this is not the place we want to do that.

I would take it as a healthy input is one that will report events when they exist. So if the input cannot even read a directory, or if it cannot connect to AWS then it should not be reported as healthy.